src/v/raft/consensus.cc

```diff
@@ -1952,8 +1957,20 @@ ss::future<> consensus::do_hydrate_snapshot(storage::snapshot_reader& reader) {
               return _offset_translator.prefix_truncate_reset(
                 _last_snapshot_index, delta);
           })
-          .then([this] { return truncate_to_latest_snapshot(); })
-          .then([this] { _log.set_collectible_offset(_last_snapshot_index); });
+          .then([this] {
+              if (
+                ntp() == model::controller_ntp
```

Comment:
please make it configurable from outside, we do not want to be ntp specific in our raft implementation